Assets/Scripts/Character/Ability/AdvancedAbilities/GrenadeSpawn.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GrenadeSpawn : MonoBehaviour
+{
+    private float throw_power = 5f;
+    public GameObject Grenade;
+    private GameObject characterParent;
+    // Start is called before the first frame update
+    void Start()
+    { 
+        characterParent = transform.parent.gameObject;   
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    public void CreateGrenade(Vector3 direction) {
```

Comment:
move this function into GrenadeAbility and if you want to use any monobehavior functions do character.###
where ### is what you want from monobehavior